### PR TITLE
[WIP]: Support Linux

### PR DESCRIPTION
**Pull request by harlanhaskins**
This adds support for Linux mostly by conditionally removing imports of `Accounts` and `Social`, and avoiding bridging conversions between `NSData`/`Data` and `NSString`/`String`

It's really untested, though. My twitter bot works, but that's about all the testing I've done.

### CODE REVIEW

**Comment by meteochu** [comment]
Thanks for the PR. I noticed that for Linux support, you've added many `#if os(macOS)` around the code; however, please do keep in mind that this project also supports iOS (and its variants). 

Since there are quite a few places where `#if os(macOS)` needs to become `#if os(macOS) || os(iOS)`, I'm just going to make a general note here instead of commenting on each individual one in the code review.

**Comment by harlanhaskins** [comment]
@meteochu Good catch! Thanks!

I'd like if we could set up CI on this project for macOS, iOS and Linux. I think that'd be really useful.

**Comment by harlanhaskins** [comment]
@meteochu I updated the build configurations.

File: Sources/String++.swift [hunk 1]
```diff
@@ -79,3 +77,30 @@ extension String {
 
 }
 
+
+/// Needed for API compatibility with Scanner in swift-corelibs-foundation
+/// These are exactly the same API as in 
+/// https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/NSScanner.swift
+extension Scanner {
```
Comment (meteochu) [inline Sources/String++.swift]: For the sake of project uniformity, please have the `extension Scanner { ... }` as its own file.

**Comment by meteochu** [comment]
@harlanhaskins I think setting up CI would be a good idea to test on all 3 platforms; let's see if @mattdonnelly agrees.

File: Sources/String++.swift (hunk 1 above)
Comment (harlanhaskins) [inline Sources/String++.swift]: 👍

**Pull request closed without merging**